libcontainer/cgroups/utils.go

```diff
@@ -326,9 +326,12 @@ func PathExists(path string) bool {
 func EnterPid(cgroupPaths map[string]string, pid int) error {
 	for _, path := range cgroupPaths {
 		if PathExists(path) {
-			if err := ioutil.WriteFile(filepath.Join(path, "cgroup.procs"),
-				[]byte(strconv.Itoa(pid)), 0700); err != nil {
-				return err
+			// Dont attach any pid to the cgroup if -1 is specified as a pid
+			if pid != -1 {
+				if err := ioutil.WriteFile(filepath.Join(path, "cgroup.procs"),
```

Comment:
While replacing this, maybe we should be using `writeFile` here instead of `ioutil`.